metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/BaselineRuleEvaluator.java

```diff
@@ -0,0 +1,194 @@
+    this.queryRequestHandler = queryRequestHandler;
+  }
+
+  Optional<NotificationEvent> evaluateRule(
+      MetricAnomalyEventCondition metricAnomalyEventCondition, AlertTask alertTask)
+      throws IOException {
+    BaselineThresholdCondition dynamicThresholdCondition =
+        metricAnomalyEventCondition
+            .getViolationConditionList()
+            .get(0)
+            .getBaselineThresholdCondition();
+
+    Duration duration = java.time.Duration.parse(dynamicThresholdCondition.getBaselineDuration());
+    long durationMillis = duration.toMillis();
+
+    // this query will fetch metric data, which will used for baseline calculation and evaluation
+    QueryRequest queryRequest =
+        queryRequestHandler.getQueryRequest(
+            metricAnomalyEventCondition,
+            alertTask.getTenantId(),
+            alertTask.getCurrentExecutionTime() - durationMillis,
+            alertTask.getCurrentExecutionTime());
+
+    LOGGER.debug("Rule id {}, Query request {}", alertTask.getEventConditionId(), queryRequest);
+
+    Iterator<ResultSetChunk> iterator =
+        queryRequestHandler.executeQuery(
+            Map.of(TENANT_ID_KEY, alertTask.getTenantId()), queryRequest);
+
+    List<Double> metricValuesForBaseline = new ArrayList<>();
+    List<Double> metricValuesForEvaluation = new ArrayList<>();
+
+    iterator.forEachRemaining(
+        chunk -> {
+          for (Row row : chunk.getRowList()) {
+            if (row.getColumnCount() >= 2
+                && row.getColumn(1).getValueType()
+                    == org.hypertrace.core.query.service.api.ValueType.STRING) {
+              double value = Double.parseDouble(row.getColumn(1).getString());
+              metricValuesForBaseline.add(value);
+              if (Long.parseLong(row.getColumn(0).getString())
+                  >= alertTask.getLastExecutionTime()) {
+                metricValuesForEvaluation.add(value);
+              }
+            }
+          }
+        });
+
+    LOGGER.debug(
+        "Rule id {}, Metric data for baseline {}, evaluation {}",
+        alertTask.getEventConditionId(),
+        metricValuesForBaseline,
+        metricValuesForEvaluation);
+
+    Baseline baseline =
+        getBaseline(metricValuesForBaseline.stream().mapToDouble(Double::doubleValue).toArray());
+
+    LOGGER.debug("Rule id {}, Baseline value {}", alertTask.getEventConditionId(), baseline);
+
+    int dataCount = 0, violationCount = 0;
+    for (Double metricValue : metricValuesForEvaluation) {
+      dataCount++;
+      if (metricValue < baseline.getLowerBound().getDouble()
+          || metricValue > baseline.getUpperBound().getDouble()) {
+        violationCount++;
+      }
+    }
+
+    LOGGER.debug(
+        "Rule id {}, DataCount {}, ViolationCount {}",
+        alertTask.getEventConditionId(),
+        dataCount,
+        violationCount);
+
+    if (!StaticRuleEvaluator.isViolation(dataCount, violationCount)) {
```

Comment:
nit: `isViolation` can be in a helper class or common method for readability so both static + baseline evaluator can use. This may confuse the mix of static + baseline.